In [5]:
import pandas as pd

titanic_df = pd.read_csv('assets/titanic_passengers.csv')

titanic_df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [7]:
third_class = titanic_df[titanic_df['Pclass'] == 3]

third_class.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S
5,6,0,3,"Moran, Mr. James",male,NaN,0,0,330877,8.4583,NaN,Q
7,8,0,3,"Palsson, Master. Gosta Leonard",male,2.0,3,1,349909,21.0750,NaN,S


In [9]:
third_class.groupby('Survived')['Age'].mean()

Survived
0    26.555556
1    20.646118
Name: Age, dtype: float64

In [11]:
titanic_df['Fare in CAD'] = titanic_df['Fare'] * 1.70

titanic_df[['Fare', 'Fare in CAD']].head()

,Fare,Fare in CAD
0,7.2500,12.32500
1,71.2833,121.18161
2,7.9250,13.47250
3,53.1000,90.27000
4,8.0500,13.68500


In [13]:
pd.crosstab(titanic_df['Pclass'], titanic_df['Survived'])

Survived,0,1
Pclass,,
1,80,136
2,97,87
3,372,119


Based on the crosstab results, passengers in 3rd class were the least likely to survive the Titanic voyage, with 372 deaths compared to 119 survivors.

In [16]:
titanic_df['Port_name'] = titanic_df['Embarked'].map({
    'S': 'Southampton',
    'C': 'Cherbourg',
    'Q': 'Queenstown'
})

titanic_df[['Embarked', 'Port_name']].head()

,Embarked,Port_name
0,S,Southampton
1,C,Cherbourg
2,S,Southampton
3,S,Southampton
4,S,Southampton
